### DISCUSSION TITLE
Labels added to namespace not being deployed

### DISCUSSION THREAD

**Discussion by ohheyrj** [fluxcd/flux2#3655] [Q&A] [answered]
Hello,

I previously used Flux to deploy a namespace into my cluster. I now wanted to add a new label of `goldilocks.fairwinds.com/enabled: true` to the namespace, however, once I updated the code and pushed to git, the namespace doesn't seem to deploy the changed labels.

My code (stored under cluster/production/namespaces) looks like this:
```
---
apiVersion: v1
kind: Namespace
metadata:
  name: traefik
  labels:
    goldilocks.fairwinds.com/enabled: true
```

Doing a `kubectl describe namespace traefik` gives me this:

```
Name:         traefik
Labels:       kubernetes.io/metadata.name=traefik
              kustomize.toolkit.fluxcd.io/name=flux-system
              kustomize.toolkit.fluxcd.io/namespace=flux-system
Annotations:  <none>
Status:       Active

No resource quota.

No LimitRange resource.
```

I have also tried doing:
```
flux suspend kustomization -n flux-system flux-system
flux resume kustomization -n flux-system flux-system
```

But this doesn't seem to change the outcome.  Any help would be greatly appreciated.

**Comment by stefanprodan** [comment]
If you do `flux trace namespace/traefik` does it show the Git SHA for the commit you pushed with the labels addition? Also can you please post here `kubectl get namespace traefik --show-managed-fields -oyaml`.

**Reply by ohheyrj** [reply]
I created a new namespace called `test-namespace`. I tried without the label to start, then commited the change with the additional label (similar to above). The addition of the label was done in commit SHA 2c52a163. The output from the commands are as follows:
```
$ flux trace namespace/test-namespace
Object:          Namespace/test-namespace
Status:          Managed by Flux
---
Kustomization:   flux-system
Namespace:       flux-system
Path:            ./cluster/production
Revision:        main/2c52a1639867618483fd8ce5567ba036fef3e78f
Status:          Last reconciled at 2023-03-04 21:39:00 +0000 GMT
Message:         Applied revision: main/2c52a1639867618483fd8ce5567ba036fef3e78f
---
GitRepository:   flux-system
Namespace:       flux-system
URL:             ssh://git@github.com/systemsmystery/flux-sm-main-eks-cluster
Branch:          main
Revision:        main/2c52a1639867618483fd8ce5567ba036fef3e78f
Status:          Last reconciled at 2023-03-04 21:38:57 +0000 GMT
Message:         stored artifact for revision 'main/2c52a1639867618483fd8ce5567ba036fef3e78f'
```

And the output of the second command:
```
$ kubectl get namespace test-namespace --show-managed-fields -oyaml
apiVersion: v1
kind: Namespace
metadata:
  creationTimestamp: "2023-03-04T21:37:55Z"
  labels:
    kubernetes.io/metadata.name: test-namespace
    kustomize.toolkit.fluxcd.io/name: flux-system
    kustomize.toolkit.fluxcd.io/namespace: flux-system
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:labels:
          f:kustomize.toolkit.fluxcd.io/name: {}
          f:kustomize.toolkit.fluxcd.io/namespace: {}
    manager: kustomize-controller
    operation: Apply
    time: "2023-03-04T21:37:55Z"
  name: test-namespace
  resourceVersion: "8375030"
  uid: 956a187a-c83d-423c-9a63-8de7d87fdd49
spec:
  finalizers:
  - kubernetes
status:
  phase: Active
```

As a work around for now I have manually applied the label and followed this FAQ article (https://fluxcd.io/flux/faq/#why-are-kubectl-edits-rolled-back-by-flux) using the `--field-manager=flux-client-side-apply` flag, the labels have since stuck. I didn't do this on the test-namespace however so I still have a namespace with the issue.

**Reply by apuri1-arqit** [reply]
We've got the same issue here, labels are not being applied to namespaces, whether via patches using vars passed in from `postBuild`, or hardcoded in the kustomization.

**Comment by a-nych** [comment]
I have the same issue, with the same label! The namespace is managed by Flux, and it doesn't set the labels to the namespace. Whats worse, if I set them manually, Flux removes them after a reconcile.

Kustomize controller version: `v1.2.2`

**Comment by DevSecNinja** [comment]
Same issue and same scenario with my `goldilocks` labels.

**Reply by DevSecNinja** [reply]
This fixed it for me: https://github.com/fluxcd/flux2/issues/4603#issuecomment-1939753221

**Comment by stefanprodan** [comment, marked as answer]
You need to add make the value a string e.g. `"true"`, otherwise the value of the label is of type boolean and the Kubernetes API drops it as being invalid.

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: traefik
  labels:
    goldilocks.fairwinds.com/enabled: "true"
```
